daemon/api_snaps.go

```diff
@@ -765,3 +802,73 @@ func shouldSearchStore(r *http.Request) bool {
 
 	return false
 }
+
+func snapHoldMany(inst *snapInstruction, st *state.State) (res *snapInstructionResult, err error) {
+	if err := validateHoldTime(inst.Time); err != nil {
+		return nil, err
+	}
+
+	patchValues := map[string]interface{}{"refresh.hold": inst.Time}
+	var msg string
+	var ts *state.TaskSet
+
+	if len(inst.Snaps) == 0 {
+		ts, err = configstate.ConfigureInstalled(st, "core", patchValues, 0)
+		if err != nil {
+			return nil, err
+		}
+
+		msg = i18n.G("Held refreshes for all snaps.")
+	} else {
+		ts, err = snapstate.CreateGateRefreshesTask(st, inst.Time, inst.Snaps)
```

Comment:
I am a bit unsure whether we really need to create tasks/changes for this, as they are mostly state manipulations? we can create nop changes if needed anyway. We do that already for refresh many when there is nothing to do I think?

Reply:
The hold for a specific set of snaps doesn't need it but, since the refresh for all snaps does, this was just a way to be consistent. But you're right, refreshes can already have no tasks so that makes more sense. I'll change it